src/segmentation_a_direct.jl

```diff
@@ -9,22 +9,38 @@ Apply k-means segmentation to a gray image to isolate a cluster group representi
 
 """
 function kmeans_segmentation(
-    gray_image::Matrix{Gray{Float64}}, ice_labels::Vector{Int64}
+    gray_image::Matrix{Gray{Float64}},
+    ice_labels::Vector{Int64},
+    k::Int64=4,
+    maxiter::Int64=50,
 )::BitMatrix
-    Random.seed!(45) # this seed generates consistent clusters for the final output
-    gray_image_height, gray_image_width = size(gray_image)
-    gray_image_1d = vec(gray_image)
-    @info("Done with reshape")
+    segmented = kmeans_segmentation(gray_image; k=k, maxiter=maxiter)
 
-    ## NOTE(tjd): this clusters into 4 classes and solves iteratively with a max of 50 iterations
+    ## Isolate ice floes and contrast from background
+    nlabel = StatsBase.mode(segmented[ice_labels])
```

Comment:
Is it possible for this function to be called with an empty ice_labels vector? If so what is returned as the mode?

Reply:
No. An error would be thrown:
```julia
julia> using StatsBase: mode

julia> mode([])
ERROR: ArgumentError: mode is not defined for empty collections
Stacktrace:
 [1] mode(a::Vector{Any})
   @ StatsBase ~/.julia/packages/StatsBase/ebrT3/src/scalarstats.jl:112
 [2] top-level scope
   @ REPL[4]:1

julia>
```

Reply:
OK, that's good to know. We should take a look a few steps earlier in the flow chart to check what happens when ice_labels comes out empty. We know it happens sometimes such as when the image is fully cloudy, so I think there is already a step that addresses this.

Reply:
We could wrap it so it would return some sentinel value like `missing` or `nothing` and decide on a course of action when that happens.